ocaml/xapi/xapi_pusb.ml

```diff
@@ -109,6 +109,9 @@ let set_passthrough_enabled ~__context ~self ~value =
            List.iter (fun rf ->
              try
                if (get_sm_usb_path ~__context rf) = pusb_path then begin
+                 let vbds = Db.VDI.get_VBDs ~__context ~self:rf in
+                 if vbds <> [] then
+                   raise (Api_errors.Server_error(Api_errors.pusb_vdi_conflict, [ Ref.string_of self; Ref.string_of rf ]));
                  Xapi_vdi.forget ~__context ~vdi:rf
```

Comment:
I think function `set_passthrough_enabled` needs to be refactored. The nesting of the control flow is too deep already and `List.iter` inside `List.iter` suggests that something is not well thought out here.

More specifically, if `vbs<>[]` the `raise` will make sure that `Xapi_vdi.forget ~__context ~vdi:rf` is never reached - it that correct?

Reply:
`Xapi_vdi.forget ~__context ~vdi:rf` can be reached if a vdi is ok before raise error. Will this has any issue?

Reply:
BTW, as PUSB may have more than one vdi, once one of the vdi have vbd bound, we need to make sure the set_passthrough_enabled failed.